lib/shared/image_viewer.dart

```diff
@@ -113,6 +113,26 @@ class _ImageViewerState extends State<ImageViewer> with TickerProviderStateMixin
     );
   }
 
+  Future<void> getImageSize(String imageUrl, context) async {
+    final ByteData data = await NetworkAssetBundle(Uri.parse(imageUrl)).load(imageUrl);
+    final Uint8List bytes = data.buffer.asUint8List();
+    final decodedImage = await decodeImageFromList(bytes);
+
+    setState(() {
```

Comment:
I wonder if there's any concern about rebuilding almost right away. Maybe we could not show the image until the dimensions are loaded or something. Not saying I have an answer, but something to consider!

Reply:
done @micahmo